online-shop/src/app/auth/guards/user/user-auth.guard.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { selectIsUser, selectUser } from 'src/app/store/selectors/auth.selectors';
+import { AppState } from 'src/app/store/state/app.state';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserAuthGuard implements CanActivate {
+
+
+  constructor(private store: Store<AppState>, private router: Router) { }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.store.pipe(
+      select(selectUser),
+      map((user) => {
+        if (user) {
+          if(state.url == '/login')
```

Comment:
for comparisons, use === operator
The == operator will compare for equality after doing any necessary type conversions. The === operator will not do the conversion, so if two values are not the same type === will simply return false
![image](https://user-images.githubusercontent.com/22193346/127818625-fbdd2098-0478-4959-8a50-876b6f94e227.png)